Add docs for `{NavigateEvent,PopStateEvent}.hasUAVisualTransition` (#36705)
mdn/content: files/en-us/web/api/navigateevent/hasuavisualtransition/index.md

---
title: "NavigateEvent: hasUAVisualTransition property"
short-title: hasUAVisualTransition
slug: Web/API/NavigateEvent/hasUAVisualTransition
page-type: web-api-instance-property
status:
  - experimental
browser-compat: api.NavigateEvent.hasUAVisualTransition
---

{{APIRef("Navigation API")}}{{SeeCompatTable}}

> The **`hasUAVisualTransition`** read-only property of the {{domxref("NavigateEvent")}} interface returns `true` if the navigation has been performed a visual transition before dispatching this event by the user agent, or `false` otherwise.

Reviewer: The spec language places UA wording at the beginning, which I think is helpful:

```suggestion
The **`hasUAVisualTransition`** read-only property of the {{domxref("NavigateEvent")}} interface returns `true` if the user agent performed a visual transition for this navigation before dispatching this event, or `false` otherwise.
```

> Some user agents may provide its builtin visual transition when executing site navigations. If the site author also add a visual transition, these transitions may conflict and cause confusion for the user. The property is intended to avoid the site author and the user agent both add visual transitions, and ensure only one visual transition is executed at a time to make sure good user experience.

Reviewer: What do you think about this? Isn't the main goal skipping author transitions now instead of overriding UA transitions?

```suggestion
User agents may provide a built-in visual transition when executing site navigations. If the site author also adds a visual transition, user agent and author transitions may conflict and confuse a visitor. The property lets you detect whether a UA transition was provided so that you can skip author transitions for a better user experience.
```

Reviewer: "Authors can use this to skip their custom transition" is a good way to put it.

Author: > Isn't the main goal skipping author transitions now instead of overriding UA transitions?

Yes, I agree so.
